app/src/main/java/at/tugraz/ist/guessingwords/data/service/WordService.kt

```diff
@@ -48,4 +48,46 @@ open class WordService(private val context : Context) {
             callback.whenReady(word)
         }
     }
+
+    /**
+     * Creates an In Memory Instance for merging Word Pools
+     * Populates it with local persistent Words
+     * @param callback
+     */
+    open fun createNewMultiplayerWordPool(callback: Callback<List<Long>>) {
+        thread {
+            val localWords = GWDatabase.getInstance(context).wordDao().getAll()
+            val dbTemp = GWDatabase.getInMemoryInstance(context)
+
+            val mergedIds = dbTemp.wordDao().mergeWordsIntoDB(localWords)
+            callback.whenReady(mergedIds)
+        }
+    }
+
+    /**
+     * Merges additional Words to the In Memory Instance
+     * Remove the duplicate Words before Merging
+     * @param merging
+     * @param callback
+     */
+    @OptIn(ExperimentalStdlibApi::class)
+    open fun mergeIntoDatabase(merging: List<Word>, callback: Callback<List<Long>>) {
+        thread {
+            val db = GWDatabase.getInMemoryInstance(context)
+            val localWords = GWDatabase.getInstance(context).wordDao().getAll()
+            var mWords = merging.toMutableList();
+            localWords.forEach { lWord ->
+                mWords.remove(mWords.find { mWord -> mWord.text.lowercase() == lWord.text.lowercase() })
+            }
```

Comment:
This probably works, but is hard to read and to know what exactly happens here.
Also, mWords will be cast toLowerCase multiple times instead of once (which could be performance intensive...)

A more concise syntax could be:
```kotlin
// extract lowercase texts to a separate list only once
val localWordTexts = localWords.map { word -> word.text.toLowerCase() }
// use removeAll for clearer code
mWords.removeAll { mWord -> localWordTexts.contains(mWord.text.toLowerCase()) }
```

Note: Runtime is stil not ideal (prob. something along the lines of O(m * n) there prob. is a more efficient solution

Reply:
Btw. using a map (eg. HashMap) instead of a list for localWordTexts would decrease the runtime to O(m) because lookup in a map is O(1).